pkg/kubelet/dockershim/naming.go

```diff
@@ -92,7 +102,9 @@ func parseSandboxName(name string) (*runtimeapi.PodSandboxMetadata, error) {
 	name = strings.TrimPrefix(name, "/")
 
 	parts := strings.Split(name, nameDelimiter)
-	if len(parts) != 6 {
+	// Tolerate the random suffix.
+	// TODO(random-liu): Remove 7 field case when docker 1.11 is deprecated.
+	if len(parts) != 6 && len(parts) != 7 {
```

Comment:
If there are 2 containers `abc_randomstr1` and `abc_randomstr2`. Both of them will be returned by list functions.

This will cause 2 containers with the same `Attempt` but different `ID`s being returned. This seems fine, because we mainly rely on container id and don't use `Attempt` that much.

This will cause:
* Wrong restart count. 
* New container can't create log path.